### DISCUSSION TITLE
Use timelineVariable() as dynamic parameter in likert prompt

### DISCUSSION THREAD

**Discussion by amillner** [jspsych/jsPsych#986] [Q&A] [answered]
Hi, 

On the timeline webpage (https://www.jspsych.org/overview/timeline/), you describe a situation in which you're able to immediately return the value of the timeline variable by adding a true argument. I'm wondering whether one can do this with the prompt using the survey-likert plugin? For example,

```
prompt: function(){
                var q1 = "The name is"+jsPsych.timelineVariable('name', true)+"</p>";
                return q1;
            },
```

but it displays the text literally in the webpage. 

Is it not possible to use the timeline variable in this way for a likert prompt?

Thanks,
Alex

**Comment by jodeleeuw** [comment, marked as answer]
Hi Alex,

This is something we could do a better job with on the design side. 

The reason this doesn't work is because `prompt` isn't actually a property of the trial object. It's a nested property of `questions`. You could use a dynamic parameter to specify the entire `questions` property, and use the timeline variable inside the `prompt` part of that property.

In a future version we'll make it so that nested properties can be functions independent of their parent property.

**Reply by amillner** [reply]
Hi Josh, 

Thanks. Would you be able to point me in the right direction? I actually need multiple prompts but I'm having trouble even getting one to work. I think I can do multiple prompts if I could figure out one. 

Here's what I have: 

```
type: 'survey-likert',
questions: function()
      {var prompts = 
      '[{prompt: "How much do you like  "'+jsPsych.timelineVariable('name', true) + '"?"',name: "like", labels: like_scale}]';
      return prompts},
```
I'm not super familiar with the different uses of quotes in JS but I've tried several combinations of single and double quotes to no avail. 

If I don't get a syntax error because of some improper use of quotes, I get an error saying I must specify a value for the prompt and labels. 

Any help would be awesome. 

Thanks!
Alex

**Reply by vijaymarupudi** [reply]
@amillner The function should return an Array, like you would have typed without a function. Quotation makes it a string, which is not necessary.

```javascript
var trial = {
  type: "survey-likert",
  questions: function () {
    var prompts = [
      {
        prompt: "How much do you like  " + jsPsych.timelineVariable("name", true) + "?",
        name: "like",
        labels: like_scale,
      },
    ];
    return prompts;
  },
};

```

**Reply by amillner** [reply]
Got it. Thanks!
